codegenerator/cli/templates/static/codegen/src/eventFetching/chainWorkers/HyperFuelWorker.res

```diff
@@ -3,80 +3,97 @@ open Belt
 
 exception EventRoutingFailed
 
-type event = {
-  name: string,
-  logId?: string,
-  mint?: bool,
-  isWildcard?: bool,
+type workerConfig = {
+  eventRouter: EventRouter.t<Types.fuelEventConfig>,
+  contractByEvent: Utils.WeakMap.t<Types.fuelEventConfig, Types.fuelContractConfig>,
```

Comment:
I'm interested why you choose WeakMap over Map?

As far as I understand the advantage is if you want keys/values in your map to be garbage collectable? Isn't this a static map?

Reply:
In general, it's a little bit lighter because it doesn't have iteration methods, so I prefer using it for simple hashmaps by default. No big difference in this case.